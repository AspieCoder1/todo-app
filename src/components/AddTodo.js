import React, { Component } from 'react';

export default class AddTodo extends Component {
	render() {
		return (
			<form onSubmit={this.props.onSubmit}>
				<input
					type="text"
					name="todoToAdd"
					value={this.props.todoToAdd}
					onChange={this.props.onChange}
				/>
				{this.props.errors && this.props.errors.noTodo ? (
					<p>Please provide a todo</p>
				) : null}
				<button type="submit">Add Todo</button>
			</form>
		);
	}
}
