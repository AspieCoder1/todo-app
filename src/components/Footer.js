import React from 'react';
import moment from 'moment';

export default () => {
	return (
		<footer>
			<h3>&copy;AspieCoder {moment().year()}, all rights reserved</h3>
		</footer>
	);
};
