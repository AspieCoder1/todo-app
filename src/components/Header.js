import React from 'react';
import headerStyles from '../styles/components/header.module.scss';

export default () => (
	<div>
		<div className="container">
			<h2 className={headerStyles.title}>
				TodoApp - Manage your Todos in peace
			</h2>
		</div>
	</div>
);
