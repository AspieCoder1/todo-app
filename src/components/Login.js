import React, { Component } from 'react';
import { connect } from 'react-redux';
import classNames from 'classnames';
import { Link } from 'react-router-dom';

import { loginUser, clearErrors } from '../redux/actions/auth';
import { history } from '../utils/history';
import styles from '../styles/components/userForm.module.scss';

export class Login extends Component {
	state = {
		email: '',
		password: '',
	};

	onSubmit = e => {
		e.preventDefault();
		const user = {
			email: this.state.email,
			password: this.state.password,
		};
		this.props.loginUser(user, history);
	};

	onChange = e => {
		this.setState({ [e.target.name]: e.target.value });
	};

	componentWillUnmount() {
		this.props.clearErrors();
	}

	render() {
		return (
			<div className={styles.page}>
				<div className={styles.content}>
					<h1 className={styles.title}>Login</h1>
					<form className={styles.form} onSubmit={this.onSubmit}>
						<div className={styles.inputGroup}>
							<label className={styles.label} htmlFor="email">
								Email
							</label>
							<input
								className={classNames(styles.input, {
									[styles.inputError]:
										this.props.errors && this.props.errors.email,
								})}
								type="text"
								name="email"
								id="email"
								value={this.state.email}
								onChange={this.onChange}
								autoComplete="email"
							/>
							{this.props.errors && this.props.errors.email ? (
								<p className={styles.error}>{this.props.errors.email}</p>
							) : null}
						</div>
						<div className={styles.inputGroup}>
							<label className={styles.label} htmlFor="password">
								Password
							</label>
							<input
								className={classNames(styles.input, {
									[styles.inputError]:
										this.props.errors && this.props.errors.password,
								})}
								type="password"
								name="password"
								id="password"
								value={this.state.password}
								onChange={this.onChange}
								autoComplete="password"
							/>
							{this.props.errors && this.props.errors.password ? (
								<p className={styles.error}>{this.props.errors.password}</p>
							) : null}
						</div>
						<button className={styles.button} type="submit">
							Log In
						</button>
					</form>
					<Link className={styles.link} to="/register">
						I need an account
					</Link>
				</div>
			</div>
		);
	}
}

const mapStateToProps = state => ({
	errors: state.errors,
});

export default connect(
	mapStateToProps,
	{ loginUser, clearErrors }
)(Login);
