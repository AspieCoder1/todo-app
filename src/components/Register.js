import React, { Component } from 'react';
import { connect } from 'react-redux';
import { registerUser } from '../redux/actions/auth';
import { history } from '../utils/history';

class Register extends Component {
	state = {
		email: '',
		password: ''
	};
	onChange = e => {
		this.setState({ [e.target.name]: e.target.value });
	};

	onSubmit = e => {
		e.preventDefault();
		const newUser = {
			email: this.state.email,
			password: this.state.password
		};
		this.props.registerUser(newUser, history);
	};

	render() {
		return (
			<div>
				<h1>Register</h1>
				<form onSubmit={this.onSubmit}>
					<label htmlFor="email">email</label>
					<input
						type="text"
						name="email"
						id="email"
						value={this.state.email}
						onChange={this.onChange}
						autoComplete="email"
					/>
					{this.props.errors.email ? <p>{this.props.errors.email}</p> : null}
					<label htmlFor="password">password</label>
					<input
						type="password"
						name="password"
						id="email"
						value={this.state.password}
						onChange={this.onChange}
						autoComplete="password"
					/>
					{this.props.errors.password ? (
						<p>{this.props.errors.password}</p>
					) : null}
					<button type="submit">Submit</button>
				</form>
			</div>
		);
	}
}

const mapStateToProps = state => ({
	errors: state.errors
});

export default connect(
	mapStateToProps,
	{ registerUser }
)(Register);
