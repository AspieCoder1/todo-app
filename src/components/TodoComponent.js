import React from 'react';

export default class TodoComponent extends React.Component {
	state = {
		completed: false,
	};

	componentDidMount() {
		this.setState({
			completed: this.props.todo.completed,
		});
	}

	render() {
		const { todo, removeTodo, onToggle } = this.props;
		return (
			<div>
				<input
					type="checkbox"
					name="completed"
					id="completed"
					checked={this.state.completed}
					value={this.state.completed}
					onChange={e => {
						this.setState({
							completed: e.target.checked,
						});

						onToggle(todo, this.state.completed);
					}}
				/>
				<p>COmpleted State: {`${this.state.completed}`}</p>
				<h3>{todo.body}</h3>
				<button onClick={() => removeTodo(todo._id)}>Delete Todo</button>
			</div>
		);
	}
}
