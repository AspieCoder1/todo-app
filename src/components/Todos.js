import React, { Component } from 'react';
import TodoComponent from './TodoComponent';

export default class Todos extends Component {
	render() {
		return (
			<div>
				<h2>Todos Component</h2>
				<TodoComponent />
			</div>
		);
	}
}
