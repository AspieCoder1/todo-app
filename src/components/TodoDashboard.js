import React, { Component } from 'react';

import AddTodo from './AddTodo';
import Todos from './Todos';
import {
	addTodo,
	getTodos,
	removeTodo,
	editTodo,
} from '../redux/actions/todos';
import { logoutUser } from '../redux/actions/auth';
import { connect } from 'react-redux';
import Header from './Header';

class TodoDashboard extends Component {
	state = {
		todos: [],
		filterCompleted: false,
		filteredTodos: [],
		todoToAdd: '',
	};

	componentDidMount() {
		this.props.getTodos();
	}

	onToggle = (todo, todoState) => {
		this.editTodo(todo._id, {
			completed: todoState,
		});
	};

	editTodo = (id, todoData) => {
		this.props.editTodo(id, todoData);
	};

	static getDerivedStateFromProps(props) {
		return {
			todos: props.todo.todos,
			filteredTodos: props.todo.todos,
		};
	}

	removeTodo = id => {
		if (this.state.todos.length > 0) {
			this.props.removeTodo(id);
			this.props.getTodos();
		}
	};

	onChange = e => {
		this.setState({ [e.target.name]: e.target.value });
	};

	addTodo = e => {
		e.preventDefault();
		this.props.addTodo({ todo: this.state.todoToAdd });
		this.props.getTodos();
	};

	handleLogout = () => {
		this.props.logoutUser();
	};

	render() {
		const { loading } = this.props.todo;
		let content;
		if (loading) {
			content = <p>Loading ...</p>;
		} else {
			content = (
				<Todos
					todos={this.state.filteredTodos}
					removeTodo={this.removeTodo}
					onToggle={this.onToggle}
				/>
			);
		}
		return (
			<div>
				<Header />
				<div>
					<h1>Your Todos</h1>
					<button onClick={this.handleLogout}>Logout</button>
				</div>
				<AddTodo
					onSubmit={this.addTodo}
					ref={this.new_todo}
					onChange={this.onChange}
					todoToAdd={this.state.todoToAdd}
					errors={this.props.errors}
				/>
				{content}
			</div>
		);
	}
}

const mapStateToProps = state => ({
	todo: state.todos,
	errors: state.errors,
});

export default connect(
	mapStateToProps,
	{ addTodo, getTodos, removeTodo, editTodo, logoutUser }
)(TodoDashboard);
