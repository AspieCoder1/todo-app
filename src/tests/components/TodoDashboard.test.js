import React from 'react';
import { shallow } from 'enzyme';

import { TodoDashboard } from '../../components/TodoDashboard';

it('should render TodoDashboard correctly', () => {
	const wrapper = shallow(
		<TodoDashboard todo={{ todos: [] }} getTodos={() => undefined} />
	);

	expect(wrapper).toMatchSnapshot();
});

it('should call getTodos when component did mount', () => {
	const getTodos = jest.fn();
	shallow(<TodoDashboard todo={{ todos: [] }} getTodos={getTodos} />);

	expect(getTodos).toHaveBeenCalledTimes(1);
});
