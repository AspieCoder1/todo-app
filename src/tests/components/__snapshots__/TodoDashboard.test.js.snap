// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TodoDashboard correctly 1`] = `
<div>
  <_default />
  <div
    className="container"
  >
    <div>
      <h1>
        Your Todos
      </h1>
      <button
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
    <AddTodo
      onChange={[Function]}
      onSubmit={[Function]}
      todoToAdd=""
    />
    <_default
      onToggle={[Function]}
      removeTodo={[Function]}
      todos={Array []}
    />
  </div>
</div>
`;
